src/constants/colours.ts

```diff
@@ -19,3 +19,30 @@ export const BORDER_COLOURS = {
     orange: 'border-orange border-2',
     purple: 'border-purple border-2',
 } as const satisfies Record<Colour, string>;
+
+export const TECH_COLORS: { [key: string]: string } = {
+    'Discord.py': '#7387CE',
+    FastAPI: '#F6BF00',
+    'Next.js': '#B17CA6',
+    Python: '#64B550',
+    React: '#ED8C9B',
+    Supabase: '#5DBBB5',
+    'Tailwind CSS': '#907FC3',
+    TypeScript: '#ACCB00',
+};
+
+/**
+ * Calculates the perceived luminance and returns black or white as the contrast color. Based on
+ * code from matfin. Source: https://stackoverflow.com/a/44615197/24033621, licensed under CC BY-SA
+ * 3.0.
```

Comment:
I think just having the link to the source is sufficient attribution, according to this: https://stackoverflow.blog/2009/06/25/attribution-required/

I'd probably note that it takes in a string that represents a hex color as well